monitor/monitor_test.go

```diff
@@ -0,0 +1,177 @@
+package monitor
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log"
+	"os"
+	"testing"
+	"time"
+
+	ct "github.com/google/certificate-transparency-go"
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/ct-woodpecker/test"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	logKey = "MFkwEwYHKoZIzj0CAQYIKoZIzj0DAQcDQgAElgyN7ptarCAX5krBwDwjhHM+b0xJjCKke+Dfr3GWSbLm3eO7muXRo8FDDdpdiRpnG4NJT0bdzq5YEer4C2eZ+g=="
+)
+
+func TestNew(t *testing.T) {
+	l := log.New(os.Stdout, "", log.LstdFlags)
+	clk := clock.NewFake()
+	clk.Set(time.Now())
+
+	logURI := "test"
+	processedLogKey := fmt.Sprintf("-----BEGIN PUBLIC KEY-----\n%s\n-----END PUBLIC KEY-----", logKey)
+	fetchDuration := time.Second
+
+	// Creating a monitor with an illegal key should fail
+	_, err := New(logURI, "⚷", fetchDuration, l, clk)
+	if err == nil {
+		t.Errorf("Expected New() with invalid key to error")
+	}
+
+	// Creating a monitor with vaild configuration should not fail
+	m, err := New(logURI, logKey, fetchDuration, l, clk)
+	if err != nil {
+		// NOTE: Using fatalf to avoid later nil pointer panics
```

Comment:
I'd omit this NOTE and the one below, since this is the common usage of fatalf: avoiding probelms later in the test if setup fails.

Reply:
Fixed in 575a04d